client/apikey/apikey.go

```diff
@@ -126,53 +90,25 @@ func (k *ApiKey) Create(post Post) (apikey Data, err error) {
 		err = errors.Errorf("unsuccessful, got status code %q with response: %+v for request to %s", response.Status, response.Request, path)
 		return
 	}
-	_, err = ioutil.ReadAll(response.Body)
-	if err != nil {
-		return
-	}
-	responseJSON, err = getAll(k)
+	responseData, err := ioutil.ReadAll(response.Body)
 	if err != nil {
 		return
 	}
-	apikey, ok = getOKname(post.Name, responseJSON)
-	if !ok {
-		err = errors.Errorf("Could not get key after creation: %s", apikey.Name)
-	}
-	log.Printf("[APIKEY|POST] created a new apikey %#v", apikey)
-	return
-}
-
-func (k *ApiKey) Update(post Post) (updatedApiKey Data, err error) {
-	log.Printf("[APIKEY|UPDATE] updating apikey")
-	updatedApiKey, err = k.Create(Post{})
+	err = json.Unmarshal(responseData, &responseJSON)
 	if err != nil {
 		return
 	}
-	log.Printf("[APIKEY|UPDATE] updated apikey")
 	return
 }
 
-// Delete will disable the apikey and then delete it
-func (k *ApiKey) Delete(id string) (err error) {
-	apikey, err := k.Get(id)
-	emptyKey := Data{}
-	if apikey == emptyKey {
-		return
-	}
-	log.Printf("[APIKEY|DELETE] deleting apikey with id %s", apikey.ID)
-	path := fmt.Sprintf("api/experimental/v2/api_key/%s", apikey.ID)
-	myUrl, err := url.Parse(path)
-	if err != nil {
-		return
-	}
-	resp, err := k.restClient.DoDelete(myUrl.String())
-	if err != nil {
-		return
+func findByName(name string, responseJSON Response) (apikey Data, err error) {
```

Comment:
`Data` is somewhat vague here. Maybe rename model to `APIKeyData` or something more descriptive.

Reply:
I agree this need to be addressed in the structs for the SDK